frontend/app/services/MemberService.scala

```diff
@@ -178,7 +178,9 @@ trait MemberService extends LazyLogging with ActivityTracking {
             deliveryPostcode = form.map(_.deliveryAddress.postCode),
             billingPostcode = form.flatMap(f => f.billingAddress.map(_.postCode)).orElse(form.map(_.deliveryAddress.postCode)),
             subscriptionPaymentAnnual = Some(annual),
-            marketingChoices = None
+            marketingChoices = None,
+            city = form.map(_.deliveryAddress.town),
+            country = form.map(_.deliveryAddress.country.name)
```

Comment:
Looks like you have added this to `membershipUpgrade` (which might be useful?) but you need to add this to where we track `membershipRegistration`